packages/astro/components/Picture.astro

```diff
@@ -99,7 +100,7 @@ if (import.meta.env.DEV) {
 			return (
 				<source
 					srcset={srcsetAttribute}
-					type={'image/' + image.options.format}
+					type={mime.lookup(image.options.format ?? '') ?? `image/${image.options.format}`}
```

Comment:
Instead of `''`, you can pass the src of the image. mrmime should be able to handle it

Reply:
Ok! That was to appease TypeScript, since `lookup`'s definition only accepts a string, but I've confirmed that mrmime correctly returns `undefined` if given `undefined`.

Reply:
Oh I see pass the source, read too quickly.

Reply:
Hmm passing `src` works in production builds, but in dev `mrmime` doesn't know what to make of the src URL:

```ts
import * as mrmime from 'mrmime'

// Build
mrmime.lookup('http://localhost:4321/_astro/penguin1.wSAOfMN3_ZYmXno.jpg')
// image/jpeg


// Dev
mrmime.lookup('http://localhost:4321/_image?href=%2F%40fs%2FUsers%2Fmika%2FCode%2Fastro%2Fpackages%2Fastro%2Ftest%2Ffixtures%2Fcore-image%2Fsrc%2Fassets%2Fpenguin1.jpg%3ForigWidth%3D207%26origHeight%3D243%26origFormat%3Djpg&f=jpg') 
// undefined
```

I'm not sure how common an undefined `image.options.format` is? Presumably not very since it wasn't handled in the previous implementation?

Reply:
Pushed a commit with this, just noting the above caveat. It doesn't materially change the undefined edge case.